crates/red_knot_python_semantic/src/types.rs

```diff
@@ -568,58 +575,76 @@ impl<'db> Type<'db> {
 
             (Type::None, Type::Instance(class_type)) | (Type::Instance(class_type), Type::None) => {
```

Comment:
I'm thinking to go over all of the instance of `Instance(class_type)` and rename to `Instance(instance)` so it's not misleading.